xrviz/fields.py

```diff
@@ -0,0 +1,75 @@
+import panel as pn
+import xarray as xr
+from .sigslot import SigSlot
+
+
+class Fields(SigSlot):
+    """
+    This section provides the user with a fields selection panel.
+    Upon selection of a variable in `Display` panel, its data_coords
+    are available as options in `x` and `y` Select widget.
+
+    Parameters
+    ----------
+    data: `xarray` instance: `DataSet` or `DataArray`
+           datset is used to initialize.
+
+    Attributes
+    ----------
+    x: `panel.widget.Select` for selection of coords along x-axis.
+    y: `panel.widget.Select` for selection of coords along y-axis.
+    kwargs: provides access to coords selected by `x` and `y`.
+    """
+
+    def __init__(self, data):
+        super().__init__()
+        self.set_data(data)
+        self.x = pn.widgets.Select(name='x')
+        self.y = pn.widgets.Select(name='y')
+        self.index_selectors = pn.Column()
+
+        self._register(self.x, 'x')
+
+        self.connect('x', self.change_y)
+
+        self.panel = pn.Row(pn.Column(self.x, self.y),
+                            pn.Spacer(width=200),
+                            self.index_selectors, name='Fields',)
+
+        if isinstance(data, xr.DataArray):
+            self.setup(data)
+
+    def set_data(self, data):
+        if isinstance(data, xr.Dataset) or isinstance(data, xr.DataArray):
```

Comment:
This should never be False

Reply:
Oh yes. I have shifted `set_data` to dashboard.py where this would be checked while initialisation.